demos/simple_pipeline/demo_setup/README.md

```diff
@@ -0,0 +1,183 @@
+  --project ${DEMO_PROJECT} \
+  --zone ${ZONE} \
+  --cluster-version 1.11.2-gke.9 \
+  --num-nodes=8 \
+  --scopes cloud-platform,compute-rw,storage-rw \
+  --verbosity error
+
+# scale down cluster to 3 (initial 8 is just to prevent master restarts due to upscaling)
+# we cannot use 0 because then cluster autoscaler treats the cluster as unhealthy.
+# Also having a few small non-gpu nodes is needed to handle system pods
+gcloud container clusters resize ${CLUSTER} \
+  --project ${DEMO_PROJECT} \
+  --zone ${ZONE} \
+  --size=3 \
+  --node-pool=default-pool
+
+# enable node auto provisioning
+gcloud beta container clusters update ${CLUSTER} \
+  --project ${DEMO_PROJECT} \
+  --zone ${ZONE} \
+  --enable-autoprovisioning \
+  --max-cpu 20 \
+  --max-memory 200 \
+  --max-accelerator=type=nvidia-tesla-k80,count=8
+```
+
+Once the cluster has been created, install GPU drivers:
+
+```
+kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/daemonset.yaml
+```
+
+Add RBAC permissions, which allows your user to install kubeflow components on
+the cluster:
+
+```
+kubectl create clusterrolebinding cluster-admin-binding-${USER} \
+  --clusterrole cluster-admin \
+  --user $(gcloud config get-value account)
+```
+
+Setup kubectl access:
+
+```
+kubectl create namespace kubeflow
+./create_context.sh gke ${NAMESPACE}
+```
+
+Setup OAuth environment variables ${CLIENT_ID} and ${CLIENT_SECRET} using the
+instructions
+[here](https://www.kubeflow.org/docs/started/getting-started-gke/#create-oauth-client-credentials).
+
+```
+kubectl create secret generic kubeflow-oauth --from-literal=client_id=${CLIENT_ID} --from-literal=client_secret=${CLIENT_SECRET}
+```
+
+Create service accounts, add permissions, download credentials, and create secrets:
+
+```
+ADMIN_EMAIL=${CLUSTER}-admin@${PROJECT}.iam.gserviceaccount.com
+USER_EMAIL=${CLUSTER}-user@${PROJECT}.iam.gserviceaccount.com
+ADMIN_FILE=${HOME}/.ssh/${ADMIN_EMAIL}.json
+USER_FILE=${HOME}/.ssh/${ADMIN_EMAIL}.json
+
+gcloud iam service-accounts create ${CLUSTER}-admin --display-name=${CLUSTER}-admin
+gcloud iam service-accounts create ${CLUSTER}-user --display-name=${CLUSTER}-user
+
+gcloud projects add-iam-policy-binding ${PROJECT} \
+  --member=serviceAccount:${ADMIN_EMAIL} \
+  --role=roles/storage.admin
+gcloud projects add-iam-policy-binding ${PROJECT} \
+  --member=serviceAccount:${USER_EMAIL} \
+  --role=roles/storage.admin
+
+gcloud iam service-accounts keys create ${ADMIN_FILE} \
+  --project ${PROJECT} \
+  --iam-account ${ADMIN_EMAIL}
+gcloud iam service-accounts keys create ${USER_FILE} \
+  --project ${PROJECT} \
+  --iam-account ${USER_EMAIL}
+
+kubectl create secret generic admin-gcp-sa \
+  --from-file=admin-gcp-sa.json=${ADMIN_FILE}
+kubectl create secret generic user-gcp-sa \
+  --from-file=user-gcp-sa.json=${USER_FILE}
+```
+
+Install kubeflow with the following commands:
+
+```
+kfctl init ${CLUSTER} --platform gcp
```

Comment:
Just curious - why not just use kfctl to create the GKE cluster?

Reply:
This demo highlights autoprovisioning, which is a beta feature not included in kfctl or click-to-deploy. It also includes pipelines, which needs a bit of work on access permissions in order to be included as part of kfctl.

Reply:
The plan for 0.4.0 is to include pipelines by default in kubeflow deployments, so hopefully this will simplify the process.

Reply:
can't wait for that day 💃